src/main/java/frc/robot/subsystems/Climber2022.java

```diff
@@ -0,0 +1,216 @@
+import frc.robot.motors.MotorControllerEncoder;
+import frc.robot.motors.MotorControllerFactory;
+import frc.robot.motors.MotorType;
+
+import frc.robot.tuning.SubsystemTuner;
+import java.util.Map;
+import org.apache.logging.log4j.Logger;
+
+public class Climber2022 extends SubsystemTuner {
+
+  // TODO make climber into state space and use real position control, after granite state
+  private final MotorControllerEncoder climberMotorLeft = MotorControllerFactory.create(
+      RobotConstants.get().climber2022LeftMotorId(), MotorType.SPARK_MAX_BRUSHLESS);
+  private final MotorControllerEncoder climberMotorRight = MotorControllerFactory.create(
+      RobotConstants.get().climber2022RightMotorId(), MotorType.SPARK_MAX_BRUSHLESS);
+  private final Relay climberLock = new Relay(RobotConstants.get().climber2022SolenoidChannel());
+
+  private boolean enabled = false;
+  private static final Logger LOGGER = RobotLogManager.getMainLogger(Climber2022.class.getName());
+
+
+  public Climber2022() {
+    super();
+
+    climberMotorLeft.setInverted(RobotConstants.get().climber2022LeftMotorInverted());
+    climberMotorRight.setInverted(RobotConstants.get().climber2022RightMotorInverted());
+    climberLock.set(Value.kOff);
+  }
+
+  public void enable() {
+    climberLock.set(Value.kReverse);
+    enabled = true;
+  }
+
+  public void disable() {
+    climberLock.set(Value.kOff);
+    stop();
+    enabled = false;
+  }
+
+  public boolean isEnabled() {
+    return enabled;
+  }
+
+  public void upLeft() {
+    if (enabled) {
+      if (climberMotorLeft.getPosition() < RobotConstants.get().climber2022LeftUpperLimit()) {
+        climberMotorLeft.set(RobotConstants.get().climber2022UpSpeed());
+      } else {
+        stopLeft();
+      }
+    }
+  }
+
+  public void upRight() {
+    if (enabled) {
+      if (climberMotorRight.getPosition() < RobotConstants.get().climber2022RightUpperLimit()) {
+        climberMotorRight.set(RobotConstants.get().climber2022UpSpeed());
+      } else {
+        stopRight();
+      }
+    }
+  }
+
+  public void up() {
+    upLeft();
+    upRight();
+  }
+
+  public void downSafeLeft() {
+    if (enabled) {
+      if (climberMotorLeft.getPosition() > RobotConstants.get().climber2022LeftLowerLimit()) {
+        climberMotorLeft.set(-RobotConstants.get().climber2022DownSpeed());
+      } else {
+        stopLeft();
+      }
+    }
+  }
+
+  public void downSafeRight() {
+    if (enabled) {
+      if (climberMotorRight.getPosition() > RobotConstants.get().climber2022RightLowerLimit()) {
+        climberMotorRight.set(-RobotConstants.get().climber2022DownSpeed());
+      } else {
+        stopRight();
+      }
+    }
+  }
+
+  public void downSafe() {
+    downSafeLeft();
+    downSafeRight();
+  }
+
+  public void downFullLeft() {
+    if (enabled) {
+      if (climberMotorLeft.getPosition() > 2) {
+        climberMotorLeft.set(-RobotConstants.get().climber2022DownSpeed());
+      } else {
+        stopLeft();
+      }
+    }
```

Comment:
I'd like a comment on these functions. What does `2` mean? Where does it come from?